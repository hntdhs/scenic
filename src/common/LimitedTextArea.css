#comment-box {
    max-width: 50%;
}

.text-area p {
    font-size: medium;
}