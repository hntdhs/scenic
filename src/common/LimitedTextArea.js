import React, { useState, useCallback, useEffect } from "react";


const LimitedTextArea = ({ value, limit, onChange, domId }) => {
  const [content, setContent] = useState("");

  const setFormattedContent = useCallback(
    text => {
      // const limitedContent = text.slice(0, limit)
      const limitedContent = text.slice(0, limit)
      setContent(limitedContent);
      onChange(limitedContent);
    },
    [limit, setContent ]
  );
  // useCallback is used to isolate resource intensive functions so that they won't automatically run on every render; it only runs when one of its dependencies update - I thought useEffects run every time one of its dependencies updates? What's the difference?

  useEffect(() => {
    setFormattedContent(value)
  }, [value])

  return (
    <div>
      <textarea
        id={domId}
        onChange={event => setFormattedContent(event.target.value)}
        value={content}
      />
      <p>
        Maximum length: {limit}<br></br>
        {content.length}/{limit}
      </p>
    </div>
  );
}

export default LimitedTextArea;