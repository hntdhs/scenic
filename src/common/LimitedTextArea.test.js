import React from "react";
import { render } from "@testing-library/react";
import LimitedTextArea from "./LimitedTextArea.js"
import { expect } from "@jest/globals";

it("matches snapshot", function () {
    const testHandlerMock = jest.fn(x => 42 + x);
    const { asFragment } = render(<LimitedTextArea value="Just some text" onChange={testHandlerMock} />);
    expect(testHandlerMock).toBeCalled()
  });