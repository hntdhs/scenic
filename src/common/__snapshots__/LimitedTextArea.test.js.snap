// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <textarea>
      Just some text
    </textarea>
    <p>
      Maximum length: 
      <br />
      14/
    </p>
  </div>
</DocumentFragment>
`;
